lib/plugins/cannulaage.js

```diff
@@ -3,53 +3,110 @@
+    
+    sbx.notifications.requestNotify({
+      level: 2
+      , title: title
+      , message: title + msg
+      , pushoverSound: sound
+      , plugin: cage
+      , debug: {
+        cannulaInfo: cannulaInfo
+      }
+    });
 
-  function cage() {
-    return cage;
   }
+}
+
+function frac(f) {
+  return f % 1;
+}
+
+cage.findLatestTimeChange = function findLatestTimeChange(sbx) {
+
+  var returnValue = {'message':'', 'found': false, 'age': 0, 'treatmentDate': null, 'checkForAlert': false};
+
+  var prevDate = new Date('1900-01-01');
 
-  cage.label = 'Cannula Age';
-  cage.pluginType = 'pill-minor';
-
-  cage.updateVisualisation = function updateVisualisation (sbx) {
-    var age = 0;
-    var found = false;
-    var treatmentDate = null;
-    var message = '';
-
-    _.forEach(sbx.data.treatments, function eachTreatment (treatment) {
-      if (treatment.eventType === 'Site Change') {
-        treatmentDate = new Date(treatment.created_at);
-        var hours = Math.round(Math.abs(sbx.time - treatmentDate) / 36e5);
-
-        if (!found) {
-          found = true;
-          age = hours;
-        } else {
-          if (hours < age) {
-            age = hours;
-            if (treatment.notes) {
-              message = treatment.notes;
-            }
+  _.forEach(sbx.data.treatments, function eachTreatment (treatment) {
+    var treatmentDate = new Date(treatment.created_at);
+    if (treatment.eventType === 'Site Change' && treatmentDate > prevDate && treatmentDate < sbx.time) {
+    
+      prevDate = treatmentDate;
+      returnValue.treatmentDate = treatmentDate;
+
+      //allow for 10 minute period after a full hour during which we'll alert the user
+      var a = moment(sbx.time);
+      var b = moment(returnValue.treatmentDate);
+      var hours = a.diff(b,'hours');
+      var minFractions = a.diff(b,'minutes') - hours * 60;
+      
+      returnValue.checkForAlert = minFractions <= 10 ? true : false;
+      
+      if (!returnValue.found) {
+        returnValue.found = true;
+        returnValue.age = hours;
+      } else {
+        if (hours > 0 && hours < returnValue.age) {
+          returnValue.age = hours;
+          if (treatment.notes) {
+            returnValue.message = treatment.notes;
           }
         }
       }
-    });
+    }
+  });
 
-    var info = [{label: 'Inserted:', value: moment(treatmentDate).format('lll')}];
-    if (message !== '') { info.push({label: 'Notes:', value: message}); }
+  return returnValue;
+}
 
-    sbx.pluginBase.updatePillText(cage, {
-      value: age + 'h'
-      , label: 'CAGE'
-      , info: info
-    });
+cage.updateVisualisation = function updateVisualisation (sbx) {
 
-  };
+  var cannulaInfo = cage.findLatestTimeChange(sbx);
 
-  return cage();
-}
+  var info = [{label: 'Inserted:', value: moment(cannulaInfo.treatmentDate).format('lll')}];
+  if (cannulaInfo.message !== '') info.push({label: 'Notes:', value: cannulaInfo.message});
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Expected '{' and instead saw 'info'.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769587957/issues/source?bid=2452379&fileBranchId=2457543#l103)